### DISCUSSION TITLE
current_user should accept optional kwargs

### DISCUSSION THREAD

**Discussion by e-kondr01** [fastapi-users/fastapi-users#1130] [Ideas]
In case you want to authenticate a user based on fields other than "is_active", "is_verified" and "is_superuser", it would be nice to be able to pass other field names as kwargs to current_user().
Example: 
`current_user = fastapi_users.current_user(is_deleted=False)`
which authentcates user only if the value of "is_deleted" field is False for them.

**Comment by hsul-git** [comment]
Im looking for the same solution.

**Comment by frankie567** [comment]
The thing is that current flag aren't just fields lookup, there is actually a complete logic involved behind them: https://github.com/fastapi-users/fastapi-users/blob/1104460533ddc661909f37c3edec017b1b5470d1/fastapi_users/authentication/authenticator.py#L174-L186

If you need to add your own logic, you could very well define your own dependency:

```py
async def current_user_not_deleted(user: User = Depends(current_active_user)):
    if user.is_deleted:
        raise HTTPException(401)
    return user
````
